From b922a937abd45de4d9585f38031b4ab081c91d36 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 15 Feb 2024 13:39:19 +0100
Subject: [PATCH 0590/1695] core: dt_driver: support DT_DRIVER_NOTYPE in
 fdt_get_dt_driver_cells()

Return a null cell size in case of DT_DRIVER_NOTYPE.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d9baed75db7c6d2d8e2253f907f50fce46c1d01c core: dt_driver: support DT_DRIVER_NOTYPE in fdt_get_dt_driver_cells()]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I7254ede1a3c5a2ba8502534a75ba24a9ebf205cc
---
 core/kernel/dt_driver.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/kernel/dt_driver.c b/core/kernel/dt_driver.c
index 67c4052fa..30dfa46cf 100644
--- a/core/kernel/dt_driver.c
+++ b/core/kernel/dt_driver.c
@@ -200,6 +200,7 @@ int fdt_get_dt_driver_cells(const void *fdt, int nodeoffset,
 		break;
 	case DT_DRIVER_I2C:
 	case DT_DRIVER_REGULATOR:
+	case DT_DRIVER_NOTYPE:
 		return 0;
 	default:
 		panic();
-- 
2.47.2

